fairscale/experimental/wgit/repo.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pathlib
+from pathlib import Path
+import sys
+from typing import Union
+
+from .pygit import PyGit
+from .sha1_store import SHA1_store
+
+
+class Repo:
+    def __init__(self, parent_dir: Path, init: bool = False) -> None:
+        """Features:
+        1. Create the wgit directory if it does not exist.
+        2. SHA1Store.init()
+        3. Create SHA1 .wgit/sha1_refs.json
+        3. Initialize a .git directory within the .wgit using `git init`.
+        4. add a .gitignore within the .wgit directory, so that the git repo within will ignore `sha1_refs.json`
+        """
+        # If repo does not exist, creates a new wgit repo object with self.repo.path pointing to the path of repo
+        # and notes all the internal files.
+        # else, if repo already exists: create a pygit object from the .wgit/.git.
+        self.wgit_parent = parent_dir
+        self._repo_path: Union[None, Path] = None
+        self._wgit_dir = Path(".wgit")
+        self._metadata_file = Path(".wgit/checkpoint.pt")
```

Comment:
I am still confused by the need of this file. I think there shouldn't be a single hardcode named metadata file like this.

Instead, we will have metadata file PER weight file users ask us to track.

Say, user have a file name `cifar.pt`, and he does:

wgit add cifar.pt

then,

inside .wgit/git, we should create/modify:

.wgit/git/cifar.pt

which is a metadata file. Am I missing anything that you are thinking otherwise?

Reply:
I think this makes sense for a more general use case. I think I was operating under the impression that we would be making users track only one checkpoint file per weigit repo. Will go ahead and include these. Thanks!!